services/QuillLMS/app/models/subscription.rb

```diff
@@ -116,12 +116,12 @@ def is_trial?
   end
 
   def check_if_purchaser_email_is_in_database
-    if purchaser_email && !purchaser_id
-      purchaser_id = User.find_by_email(purchaser_email)&.id
-      if purchaser_id
-        update(purchaser_id: purchaser_id)
-      end
-    end
+    return unless purchaser_email && !purchaser_id
+
+    purchaser_id = User.find_by_email(purchaser_email)&.id
```

Comment:
Unrelated, this is a local var with the same name as an attribute. Let's change this to something like `new_purchaser_id`